packages/gatsby-source-graphql/src/batching/dataloader-link.ts

```diff
@@ -0,0 +1,81 @@
+import DataLoader from "dataloader"
+import { ApolloLink, Observable, Operation, FetchResult } from "apollo-link"
+import { print } from "graphql"
+import { IQuery, IQueryResult, merge, resolveResult } from "./merge-queries"
+
+interface IOptions {
+  uri: string
+  fetch: Function
+  fetchOptions?: object
+  dataLoaderOptions?: object
+  headers?: object
+}
+
+export function createDataloaderLink(options: IOptions): ApolloLink {
+  const load = async (keys: ReadonlyArray<IQuery>): Promise<IQueryResult[]> => {
+    const query = merge(keys)
+    const result: object = await request(query, options)
+    if (!isValidGraphQLResult(result)) {
+      const error = extractFirstError(result)
+      throw new Error(`Failed to load query batch: ${error}`)
```

Comment:
Why only first error? I'd rather format them all (or are there too many?)

Reply:
Good point. I will change this.